### ISSUE TITLE
Bug: Can't recieved class object from json string

### ISSUE THREAD

**Issue by sonvp** [robohorse/RoboPOJOGenerator#61]
Hi @robohorse . I test it with my json below. But it show popup "incorrect structure", I tested it with http://jsonviewer.stack.hu/, It work well
This is my json.

```
[
  {
    "name": "YouTube DASH",
    "samples": [
      {
        "name": "Google Glass (MP4,H264)",
        "uri": "http://www.youtube.com/api/manifest/dash/id/bf5bb2419360daf1/source/youtube?as=fmp4_audio_clear,fmp4_sd_hd_clear&sparams=ip,ipbits,expire,source,id,as&ip=0.0.0.0&ipbits=0&expire=19000000000&signature=51AF5F39AB0CEC3E5497CD9C900EBFEAECCCB5C7.8506521BFC350652163895D4C26DEE124209AA9E&key=ik0",
        "extension": "mpd"
      },
      {
        "name": "Google Play (MP4,H264)",
        "uri": "http://www.youtube.com/api/manifest/dash/id/3aa39fa2cc27967f/source/youtube?as=fmp4_audio_clear,fmp4_sd_hd_clear&sparams=ip,ipbits,expire,source,id,as&ip=0.0.0.0&ipbits=0&expire=19000000000&signature=A2716F75795F5D2AF0E88962FFCD10DB79384F29.84308FF04844498CE6FBCE4731507882B8307798&key=ik0",
        "extension": "mpd"
      },
      {
        "name": "Google Glass (WebM,VP9)",
        "uri": "http://www.youtube.com/api/manifest/dash/id/bf5bb2419360daf1/source/youtube?as=fmp4_audio_clear,webm2_sd_hd_clear&sparams=ip,ipbits,expire,source,id,as&ip=0.0.0.0&ipbits=0&expire=19000000000&signature=249B04F79E984D7F86B4D8DB48AE6FAF41C17AB3.7B9F0EC0505E1566E59B8E488E9419F253DDF413&key=ik0",
        "extension": "mpd"
      },
      {
        "name": "Google Play (WebM,VP9)",
        "uri": "http://www.youtube.com/api/manifest/dash/id/3aa39fa2cc27967f/source/youtube?as=fmp4_audio_clear,webm2_sd_hd_clear&sparams=ip,ipbits,expire,source,id,as&ip=0.0.0.0&ipbits=0&expire=19000000000&signature=B1C2A74783AC1CC4865EB312D7DD2D48230CC9FD.BD153B9882175F1F94BFE5141A5482313EA38E8D&key=ik0",
        "extension": "mpd"
      }
    ]
  },
  {
    "name": "Widevine DASH Policy Tests (GTS)",
    "samples": [
      {
        "name": "WV: HDCP not specified",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?video_id=d286538032258a1c&provider=widevine_test"
      },
      {
        "name": "WV: HDCP not required",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?video_id=48fcc369939ac96c&provider=widevine_test"
      },
      {
        "name": "WV: HDCP required",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?video_id=e06c39f1151da3df&provider=widevine_test"
      },
      {
        "name": "WV: Secure video path required (MP4,H264)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?video_id=0894c7c8719b28a0&provider=widevine_test"
      },
      {
        "name": "WV: Secure video path required (WebM,VP9)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/vp9/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?video_id=0894c7c8719b28a0&provider=widevine_test"
      },
      {
        "name": "WV: Secure video path required (MP4,H265)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/hevc/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?video_id=0894c7c8719b28a0&provider=widevine_test"
      },
      {
        "name": "WV: HDCP + secure video path required",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?video_id=efd045b1eb61888a&provider=widevine_test"
      },
      {
        "name": "WV: 30s license duration (fails at ~30s)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?video_id=f9a34cab7b05881a&provider=widevine_test"
      }
    ]
  },
  {
    "name": "Widevine HDCP Capabilities Tests",
    "samples": [
      {
        "name": "WV: HDCP: None (not required)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?video_id=HDCP_None&provider=widevine_test"
      },
      {
        "name": "WV: HDCP: 1.0 required",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?video_id=HDCP_V1&provider=widevine_test"
      },
      {
        "name": "WV: HDCP: 2.0 required",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?video_id=HDCP_V2&provider=widevine_test"
      },
      {
        "name": "WV: HDCP: 2.1 required",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?video_id=HDCP_V2_1&provider=widevine_test"
      },
      {
        "name": "WV: HDCP: 2.2 required",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?video_id=HDCP_V2_2&provider=widevine_test"
      },
      {
        "name": "WV: HDCP: No digital output",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?video_id=HDCP_NO_DIGTAL_OUTPUT&provider=widevine_test"
      }
    ]
  },
  {
    "name": "Widevine DASH: MP4,H264",
    "samples": [
      {
        "name": "WV: Clear SD & HD (MP4,H264)",
        "uri": "https://storage.googleapis.com/wvmedia/clear/h264/tears/tears.mpd"
      },
      {
        "name": "WV: Clear SD (MP4,H264)",
        "uri": "https://storage.googleapis.com/wvmedia/clear/h264/tears/tears_sd.mpd"
      },
      {
        "name": "WV: Clear HD (MP4,H264)",
        "uri": "https://storage.googleapis.com/wvmedia/clear/h264/tears/tears_hd.mpd"
      },
      {
        "name": "WV: Clear UHD (MP4,H264)",
        "uri": "https://storage.googleapis.com/wvmedia/clear/h264/tears/tears_uhd.mpd"
      },
      {
        "name": "WV: Secure SD & HD (MP4,H264)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?provider=widevine_test"
      },
      {
        "name": "WV: Secure SD (MP4,H264)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears_sd.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?provider=widevine_test"
      },
      {
        "name": "WV: Secure HD (MP4,H264)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears_hd.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?provider=widevine_test"
      },
      {
        "name": "WV: Secure UHD (MP4,H264)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears_uhd.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?provider=widevine_test"
      }
    ]
  },
  {
    "name": "Widevine DASH: WebM,VP9",
    "samples": [
      {
        "name": "WV: Clear SD & HD (WebM,VP9)",
        "uri": "https://storage.googleapis.com/wvmedia/clear/vp9/tears/tears.mpd"
      },
      {
        "name": "WV: Clear SD (WebM,VP9)",
        "uri": "https://storage.googleapis.com/wvmedia/clear/vp9/tears/tears_sd.mpd"
      },
      {
        "name": "WV: Clear HD (WebM,VP9)",
        "uri": "https://storage.googleapis.com/wvmedia/clear/vp9/tears/tears_hd.mpd"
      },
      {
        "name": "WV: Clear UHD (WebM,VP9)",
        "uri": "https://storage.googleapis.com/wvmedia/clear/vp9/tears/tears_uhd.mpd"
      },
      {
        "name": "WV: Secure SD & HD (WebM,VP9)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/vp9/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?provider=widevine_test"
      },
      {
        "name": "WV: Secure SD (WebM,VP9)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/vp9/tears/tears_sd.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?provider=widevine_test"
      },
      {
        "name": "WV: Secure HD (WebM,VP9)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/vp9/tears/tears_hd.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?provider=widevine_test"
      },
      {
        "name": "WV: Secure UHD (WebM,VP9)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/vp9/tears/tears_uhd.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?provider=widevine_test"
      },
      {
        "name": "WV: Secure Subsample (WebM, VP9 with altref)",
        "uri": "https://storage.googleapis.com/widevine_test/vp9/sintel_1080p_vp9_altref_subsample/sintel_1080p_vp9_altref_subsample.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://widevine-proxy.appspot.com/proxy"
      },
      {
        "name": "WV: Secure Fullsample (WebM, VP9 with altref)",
        "uri": "https://storage.googleapis.com/widevine_test/vp9/sintel_1080p_vp9_altref_fullsample/sintel_1080p_vp9_altref_fullsample.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://widevine-proxy.appspot.com/proxy"
      },
      {
        "name": "WV: Secure Subsample (WebM, VP9 without altref)",
        "uri": "https://storage.googleapis.com/widevine_test/vp9/sintel_1080p_vp9_noaltref_subsample/sintel_1080p_vp9_noaltref_subsample.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://widevine-proxy.appspot.com/proxy"
      },
      {
        "name": "WV: Secure Fullsample (WebM, VP9 without altref)",
        "uri": "https://storage.googleapis.com/widevine_test/vp9/sintel_1080p_vp9_noaltref_fullsample/sintel_1080p_vp9_noaltref_fullsample.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://widevine-proxy.appspot.com/proxy"
      }
    ]
  },
  {
    "name": "Widevine DASH: MP4,H265",
    "samples": [
      {
        "name": "WV: Clear SD & HD (MP4,H265)",
        "uri": "https://storage.googleapis.com/wvmedia/clear/hevc/tears/tears.mpd"
      },
      {
        "name": "WV: Clear SD (MP4,H265)",
        "uri": "https://storage.googleapis.com/wvmedia/clear/hevc/tears/tears_sd.mpd"
      },
      {
        "name": "WV: Clear HD (MP4,H265)",
        "uri": "https://storage.googleapis.com/wvmedia/clear/hevc/tears/tears_hd.mpd"
      },
      {
        "name": "WV: Clear UHD (MP4,H265)",
        "uri": "https://storage.googleapis.com/wvmedia/clear/hevc/tears/tears_uhd.mpd"
      },
      {
        "name": "WV: Secure SD & HD (MP4,H265)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/hevc/tears/tears.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?provider=widevine_test"
      },
      {
        "name": "WV: Secure SD (MP4,H265)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/hevc/tears/tears_sd.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?provider=widevine_test"
      },
      {
        "name": "WV: Secure HD (MP4,H265)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/hevc/tears/tears_hd.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?provider=widevine_test"
      },
      {
        "name": "WV: Secure UHD (MP4,H265)",
        "uri": "https://storage.googleapis.com/wvmedia/cenc/hevc/tears/tears_uhd.mpd",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?provider=widevine_test"
      }
    ]
  },
  {
    "name": "SmoothStreaming",
    "samples": [
      {
        "name": "Super speed",
        "uri": "http://playready.directtaps.net/smoothstreaming/SSWSS720H264/SuperSpeedway_720.ism"
      },
      {
        "name": "Super speed (PlayReady)",
        "uri": "http://playready.directtaps.net/smoothstreaming/SSWSS720H264PR/SuperSpeedway_720.ism",
        "drm_scheme": "playready"
      }
    ]
  },
  {
    "name": "HLS",
    "samples": [
      {
        "name": "Apple 4x3 basic stream",
        "uri": "https://devimages.apple.com.edgekey.net/streaming/examples/bipbop_4x3/bipbop_4x3_variant.m3u8"
      },
      {
        "name": "Apple 16x9 basic stream",
        "uri": "https://devimages.apple.com.edgekey.net/streaming/examples/bipbop_16x9/bipbop_16x9_variant.m3u8"
      },
      {
        "name": "Apple master playlist advanced (TS)",
        "uri": "https://tungsten.aaplimg.com/VOD/bipbop_adv_example_v2/master.m3u8"
      },
      {
        "name": "Apple master playlist advanced (fMP4)",
        "uri": "https://tungsten.aaplimg.com/VOD/bipbop_adv_fmp4_example/master.m3u8"
      },
      {
        "name": "Apple TS media playlist",
        "uri": "https://devimages.apple.com.edgekey.net/streaming/examples/bipbop_4x3/gear1/prog_index.m3u8"
      },
      {
        "name": "Apple AAC media playlist",
        "uri": "https://devimages.apple.com.edgekey.net/streaming/examples/bipbop_4x3/gear0/prog_index.m3u8"
      },
      {
        "name": "Apple ID3 metadata",
        "uri": "http://devimages.apple.com/samplecode/adDemo/ad.m3u8"
      }
    ]
  },
  {
    "name": "Misc",
    "samples": [
      {
        "name": "Dizzy",
        "uri": "http://html5demos.com/assets/dizzy.mp4"
      },
      {
        "name": "Apple AAC 10s",
        "uri": "https://devimages.apple.com.edgekey.net/streaming/examples/bipbop_4x3/gear0/fileSequence0.aac"
      },
      {
        "name": "Apple TS 10s",
        "uri": "https://devimages.apple.com.edgekey.net/streaming/examples/bipbop_4x3/gear1/fileSequence0.ts"
      },
      {
        "name": "Android screens (Matroska)",
        "uri": "http://storage.googleapis.com/exoplayer-test-media-1/mkv/android-screens-lavf-56.36.100-aac-avc-main-1280x720.mkv"
      },
      {
        "name": "Big Buck Bunny (MP4 Video)",
        "uri": "http://redirector.c.youtube.com/videoplayback?id=604ed5ce52eda7ee&itag=22&source=youtube&sparams=ip,ipbits,expire,source,id&ip=0.0.0.0&ipbits=0&expire=19000000000&signature=513F28C7FDCBEC60A66C86C9A393556C99DC47FB.04C88036EEE12565A1ED864A875A58F15D8B5300&key=ik0"
      },
      {
        "name": "Screens 360P (WebM,VP9,No Audio)",
        "uri": "https://storage.googleapis.com/exoplayer-test-media-1/gen-3/screens/dash-vod-single-segment/video-vp9-360.webm"
      },
      {
        "name": "Screens 480p (FMP4,H264,No Audio)",
        "uri": "https://storage.googleapis.com/exoplayer-test-media-1/gen-3/screens/dash-vod-single-segment/video-avc-baseline-480.mp4"
      },
      {
        "name": "Screens 1080p (FMP4,H264, No Audio)",
        "uri": "https://storage.googleapis.com/exoplayer-test-media-1/gen-3/screens/dash-vod-single-segment/video-137.mp4"
      },
      {
        "name": "Screens (FMP4,AAC Audio)",
        "uri": "https://storage.googleapis.com/exoplayer-test-media-1/gen-3/screens/dash-vod-single-segment/audio-141.mp4"
      },
      {
        "name": "Google Play (MP3 Audio)",
        "uri": "http://storage.googleapis.com/exoplayer-test-media-0/play.mp3"
      },
      {
        "name": "Google Play (Ogg/Vorbis Audio)",
        "uri": "https://storage.googleapis.com/exoplayer-test-media-1/ogg/play.ogg"
      },
      {
        "name": "Google Glass (WebM Video with Vorbis Audio)",
        "uri": "http://demos.webmproject.org/exoplayer/glass_vp9_vorbis.webm"
      },
      {
        "name": "Google Glass (VP9 in MP4/ISO-BMFF)",
        "uri": "http://demos.webmproject.org/exoplayer/glass.mp4"
      },
      {
        "name": "Google Glass DASH - VP9 and Opus",
        "uri": "http://demos.webmproject.org/dash/201410/vp9_glass/manifest_vp9_opus.mpd"
      },
      {
        "name": "Big Buck Bunny (FLV Video)",
        "uri": "http://vod.leasewebcdn.com/bbb.flv?ri=1024&rs=150&start=0"
      }
    ]
  },
  {
    "name": "Playlists",
    "samples": [
      {
        "name": "Cats -> Dogs",
        "playlist": [
          {
            "uri": "http://html5demos.com/assets/dizzy.mp4"
          },
          {
            "uri": "http://storage.googleapis.com/exoplayer-test-media-1/mkv/android-screens-lavf-56.36.100-aac-avc-main-1280x720.mkv"
          }
        ]
      },
      {
        "name": "Audio -> Video",
        "playlist": [
          {
            "uri": "https://storage.googleapis.com/exoplayer-test-media-1/gen-3/screens/dash-vod-single-segment/audio-141.mp4"
          },
          {
            "uri": "http://storage.googleapis.com/exoplayer-test-media-1/mkv/android-screens-lavf-56.36.100-aac-avc-main-1280x720.mkv"
          }
        ]
      },
      {
        "name": "Clear -> Enc -> Clear -> Enc -> Enc",
        "drm_scheme": "widevine",
        "drm_license_url": "https://proxy.uat.widevine.com/proxy?provider=widevine_test",
        "playlist": [
          {
            "uri": "http://html5demos.com/assets/dizzy.mp4"
          },
          {
            "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears_sd.mpd"
          },
          {
            "uri": "http://html5demos.com/assets/dizzy.mp4"
          },
          {
            "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears_sd.mpd"
          },
          {
            "uri": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears_sd.mpd"
          }
        ]
      }
    ]
  }
]
```

**Comment by robohorse** [comment]
That has happened because you have array as root object. Plugin will works if you wrap array by some JSONObject like this:
```json
{"data":[...]}
```
instead of 
```json
{[...]}
```
But ok, i will try fix it)

**Comment by robohorse** [comment]
also now you can get item from array and generate POJO by plugin

**Comment by robohorse** [comment]
will be fixed in 1.8.4

**Comment by anubhavsaxena14** [comment]
While selecting target POJO type, I am getting error
![screen shot_pojo generator error](https://user-images.githubusercontent.com/9932037/48358598-17f7ea00-e6c1-11e8-8def-c49a1fa16f9c.png)
